### PR TITLE
Use redux toolkit for dependencies page

### PR DESCRIPTION
**Pull request by tacigar**
Use redux toolkit (`createSlice` and `createAsyncThunk`) for dependencies page and show AlertSnackbar when failed to load dependencies data.

スクリーンショット 2020-07-11 21 01 47

### CODE REVIEW

File: zipkin-lens/src/components/App/AlertSnackbar.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ const AlertSnackbar: React.FC = () => {
   return (
     <Snackbar
       open={alertOpen}
-      autoHideDuration={10000}
```
Comment (tacigar) [inline zipkin-lens/src/components/App/AlertSnackbar.tsx:35]: 10 seconds seemed a bit long for me.

File: zipkin-lens/src/features/dependencies/DependenciesGraph.tsx [hunk 2]
```diff
@@ -11,17 +11,26 @@
...
+
+import {
+  Box,
+  Grid,
+  Theme,
+  createStyles,
+  makeStyles,
+  useTheme,
+} from '@material-ui/core';
+import moment from 'moment';
```
Comment (tacigar) [inline zipkin-lens/src/features/dependencies/DependenciesGraph.tsx]: All import statements are sorted alphabetically.

File: zipkin-lens/src/features/dependencies/DependenciesPage.tsx [hunk 3]
```diff
@@ -132,10 +139,23 @@ const DependenciesPage: React.FC<DependenciesPageProps> = ({
...
+      dispatch(
+        setAlert({
+          message: 'Failed to load dependencies...',
+          severity: 'error',
+        }),
+      );
+    } else {
+      dispatch(clearAlert());
+    }
+  }, [error, dispatch]);
```
Comment (tacigar) [inline zipkin-lens/src/features/dependencies/DependenciesPage.tsx]: Show AlertSnackbar when failed to load dependencies.

File: zipkin-lens/src/features/dependencies/Edge.ts [hunk 4]
```diff
@@ -11,8 +11,14 @@
  * or implied. See the License for the specific language governing permissions and limitations under
  * the License.
  */
-import createReducer from '../reducers';
 
-type RootState = ReturnType<ReturnType<typeof createReducer>>;
+type Edge = {
```
Comment (tacigar) [inline zipkin-lens/src/features/dependencies/Edge.ts]: Edge was defined in `NodeDetailData.tsx`, but it is now in a separate file Transferred.

File: zipkin-lens/src/store/index.ts [hunk 5]
```diff
@@ -12,6 +12,16 @@
...
  */
 
+import { Action } from 'redux';
+import { ThunkAction } from 'redux-thunk';
+
 import createReducer from '../reducers';
 
 export type RootState = ReturnType<ReturnType<typeof createReducer>>;
+
+export type AppThunk<ReturnType = void> = ThunkAction<
```
Comment (tacigar) [inline zipkin-lens/src/store/index.ts:22]: I defined AppThunk in the same file as RootState.

**Comment by codefromthecrypt** [comment]
does this look like something that will land in the next day or two @tacigar @anuraaga ?

**Review by anuraaga — APPROVED** [review]
Nice code reduction :)

**Comment by codefromthecrypt** [comment]
Thanks @tacigar !

**Comment by tacigar** [comment]
Thanks!

**Pull request merged**